arrays/arrays.c

```diff
@@ -54,13 +54,17 @@ void destroy_array(Array *arr) {
 void resize_array(Array *arr) {
 
   // Create a new element storage with double capacity
-
+  char **new_elements = malloc(sizeof(char **) * arr->capacity * 2);
```

Comment:
this _is_ going to work because pointers are all the same size, (no matter whether it's a pointer to a value (char *, int *), or whether it's a pointer to a pointer (char **, int **). 
however, your new_elements array is an array that holds char pointers - not characters, not pointers to pointers, but pointers that themselves point to strings - so you want it to look like
`char **new_elements = malloc(sizeof(char *) * arr->capacity * 2`

let me know if I didn't explain myself too well on this point. it won't hurt your code this time, but this may bring problems in the future